### ISSUE TITLE
Source maps issues and typescript 1.8+ incompatibilities.

### ISSUE THREAD

**Issue by KirNekrasov** [angular/angular-cli#1266]
1. Windows 7
2. Versions
angular-cli: 1.0.0-beta8
node: 6.2.2
os: win32 x64
typescript: 1.8.10
chrome: 51.0.2704.103

3. Create new app => start the server => F12 in chrome => sources => no .ts files.

4. I didnt found this in known issues, so, please, add somewhere info, that current angular-cli version has this incompatibility with typescript 1.8. And for those people, who installed (or updated to) typescript 1.8: you should delete "mapRoot": "/"  from tsconfig.json and disable method fixSourceMapSources in "node_modules\angular-cli\lib\broccoli\broccoli-typescript.js", this will fix urls to your source maps in compiled .js files.

**Comment by vchatterji** [comment]
Hi,

Just adding a temporary workaround in case anyone wants to debug while this is fixed.

This is working for me for TypeScript 1.8.10 and Visual Studio Code Version 1.3.0 (1.3.0) Mac.

- Define a build task:
```
{
    // See https://go.microsoft.com/fwlink/?LinkId=733558
    // for the documentation about the tasks.json format
    "version": "0.1.0",
    "command": "tsc",
    "isShellCommand": true,
    "args": ["-p", "./src"],
    "showOutput": "silent",
    "problemMatcher": "$tsc"
}
```

- Define the launch configuration for debug:
```
        {
            "name": "Launch localhost with sourcemaps",
            "type": "chrome",
            "request": "launch",
            "url": "http://localhost:4200",
            "sourceMaps": true,
            "webRoot": "${workspaceRoot}/dist",
            "diagnosticLogging": true,
            "userDataDir": "${workspaceRoot}/.vscode/chrome"
        },
```

- Check the tsconfig.json file in the /src folder:
```
{
  "compileOnSave": false,
  "compilerOptions": {
    "declaration": false,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "module": "commonjs",
    "moduleResolution": "node",
    "noEmitOnError": true,
    "noImplicitAny": false,
    "outDir": "../dist/",
    "rootDir": ".",
    "target": "es5",
    "sourceMap": true
  },

  "files": [
    "main.ts",
    "typings.d.ts"
  ]
}
```

- Compile the project with `ng build`:
```
ng build
```
=> This will bring .html files and other assets into the `dist` directory

- Compile it again with `tsc`
```
cd src
tsc
```
=> This will update the source maps as compiled by TSC without broccoli.

- Serve the project through http-server
```
npm install http-server -g
http-server dist -p 4200
```

Launch the debug config from Visual Studio Code.

Your break points should be hit in your TS files

**Comment by filipesilva** [comment]
Closed as issue was made obsolete by #1455.
